Application/Features/CountryCQ/GetAllCoutryListQuery.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Application.DTO;
+using Application.Interfaces;
+using AutoMapper;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Application.Features.CountryCQ
+{
+    public class GetAllCoutryListQuery : IRequest <IEnumerable<CountryDto>>
+    {
+    }
+
+    public class GetAllCountryListQueryHandler : IRequestHandler<GetAllCoutryListQuery, IEnumerable<CountryDto>>
+    {
+        private readonly IApplicationDbContext _context;
+        private readonly IMapper _mapper;
+
+        public GetAllCountryListQueryHandler(IApplicationDbContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+        public async Task<IEnumerable<CountryDto>> Handle(GetAllCoutryListQuery query, CancellationToken cancellationToken)
+        {
+            var countryList = await _context.Country.ToListAsync();
```

Comment:
1 please look at ProjectAs<> method, to get from database only required fields.
2 please do not forget to pass CacellationTolen in each async method, which accepts it